### PR TITLE
Use eth-typing types

### PR DESCRIPTION
**Pull request by palango**
### What was wrong?

`eth_utils` defined own type aliases and doesn't use the ones that `eth_typing` provides. See #160 

### How was it fixed?

Used the type aliases from `eth-typing`.

I'm not sure about the formatting. When I ran `isort` it changed a lot of other files, so let me know if I should change stuff.

#### Cute Animal Picture

Cute animal picture

### CODE REVIEW

File: eth_utils/conversions.py [hunk 1]
```diff
@@ -4,7 +4,9 @@
 from .encoding import big_endian_to_int, int_to_big_endian
 from .hexadecimal import add_0x_prefix, decode_hex, encode_hex, is_hex, remove_0x_prefix
 from .types import is_boolean, is_integer, is_string
-from .typing import HexStr, Primitives, T
+from .typing import T
```
Comment (pipermerriam) [inline eth_utils/conversions.py]: I would go ahead and drop the `eth_utils.typing` module entirely and just define `T` locally here in this module.

**Comment by pipermerriam** [comment]
Good to :shipit: with that one small change.

File: eth_utils/conversions.py (hunk 1 above)
Comment (palango) [reply eth_utils/conversions.py]: Fixed.

**Comment by carver** [comment]
You mean check if any dependent libraries require `eth-typing<2`?

IIRC it wasn't a "real" backwards-incompatible change, it just turned on the exporting of types, which we've done in other projects without a major version bump. Still worth a quick check against some key libraries, but I wouldn't stress about checking each one. (unless I'm remembering something wrong about the v2 bump of eth-typing)

**Comment by pipermerriam** [comment]
@carver it's less about the "not-actually-backwards-compatible" change and more about if anything has `eth-typing<2` pinned then if we update `eth-utils` to require `eth-typing>=2.x` then those libraries will fail during installation because python packaging sucks at this stuff.  So I'd like to know that anything that's going to allow the next `eth-utils` version to install is also going to allow the `eth-typing` upgrade.

**Comment by carver** [comment]
> @carver it's less about the "not-actually-backwards-compatible" change and more about if anything has `eth-typing<2` pinned then if we update `eth-utils` to require `eth-typing>=2.x` then those libraries will fail during installation because python packaging sucks at this stuff. So I'd like to know that anything that's going to allow the next `eth-utils` version to install is also going to allow the `eth-typing` upgrade.

Absolutely. Focusing on just the packages: https://github.com/ethereum/eth-utils/network/dependents?dependent_type=PACKAGE

The very first one I looked at has the problem :/
https://github.com/ethereum/sharding/blob/master/requirements.txt

**Comment by palango** [comment]
The types used were added in 1.1 (https://github.com/ethereum/eth-typing/commit/3c6803895bb2d36a9bda39e7875884751a109ce8) , but the PEP 561 support came in 2.0 (https://github.com/ethereum/eth-typing/commit/b2fbd48d469fa6bf230ab8c2cd506e232336ea7d), so I'd update to that. I added a commit for that.

Regarding the dependencies: How do you usually handle such stuff?

**Comment by pipermerriam** [comment]
@palango I would find every library that pins `eth-utils<2` but also has `eth-typing<1` and open a pull request updating the `eth-typing` dependency to pin upper bound at `eth-typing<2` (maybe even requiring it be `>=1`.  That way we can issue releases of all of those libraries prior to merging and releasing this.

**Comment by pipermerriam** [comment]
An alternative route would be to adjust this so that we *try* to import from `eth-typing` and fall back to local versions if that fails and issue a deprecation warning.  @carver what do you think?  This is probably a more user-friendly approach and less likely to catch people by surprise due to an installation error since they should see the warning.

**Comment by carver** [comment]
> An alternative route would be to adjust this so that we _try_ to import from `eth-typing` and fall back to local versions if that fails and issue a deprecation warning. @carver what do you think? This is probably a more user-friendly approach and less likely to catch people by surprise due to an installation error since they should see the warning.

At first look, this seems promising, and faster than making sure all dependants are upgraded.

**Comment by carver** [comment]
Replaced by #163

**Pull request closed without merging**